x-pack/plugins/apm/server/routes/alerts/rule_types/get_source_fields.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AggregationsTopHitsAggregation } from '@elastic/elasticsearch/lib/api/types';
+import { SERVICE_GROUP_SUPPORTED_FIELDS } from '../../../../common/service_groups';
+
+export interface SourceDoc {
+  [key: string]: string | SourceDoc;
+}
+
+export function getSourceFieldsAgg(
+  topHitsOpts: AggregationsTopHitsAggregation = {}
+) {
+  return {
+    source_fields: {
+      top_hits: {
```

Comment:
Can you change this to `top_metrics` agg?

Reply:
With `top_metrics` we can't get all the `labels.*` fields, unless they are all enumerated in the agg. `top_hits` will get all `labels.*` without having to know all the field names in advance.